spec/features/course/assessment/question/duplication_spec.rb

```diff
@@ -6,42 +6,101 @@
 
   with_tenant(:instance) do
     let(:course) { create(:course) }
-    let(:source_assessment) do
+    let(:skill) { build(:course_assessment_skill, course: course) }
+    let(:mcq_source_assessment) do
       create(:assessment, :with_mcq_question, course: course).tap do |assessment|
         mcq_question = assessment.questions.first
-        mcq_question.question_assessments.first.skills <<
-          build(:course_assessment_skill, course: course)
+        mcq_question.question_assessments.first.skills << skill
+      end
+    end
+    let(:second_source_assessment) do
+      create(:assessment, :with_programming_question, course: course).tap do |assessment|
+        programming_question =
+          assessment.questions.where(actable_type: 'Course::Assessment::Question::Programming').first
+        programming_question.question_assessments.first.skills << skill
       end
     end
-    let(:destination_assessment) { create(:assessment, :with_programming_question, course: course) }
 
     before { login_as(user, scope: :user) }
 
     context 'As a Course Manager' do
       let(:user) { create(:course_manager, course: course).user }
+      let!(:question) { mcq_source_assessment.questions.first }
+      let!(:programming_question) do
+        questions = second_source_assessment.questions
+
+        questions.where(actable_type: 'Course::Assessment::Question::Programming').first
+      end
+
+      context 'upon duplicating a question' do
+        let!(:first_destination_assessment) { create(:assessment, course: course) }
+
+        scenario 'I can duplicate that question from one assessment to another' do
+          visit course_assessment_path(course, mcq_source_assessment)
+
+          expect(first_destination_assessment.questions.count).to be(0)
+
+          within all('section', text: question.title).first do
+            click_button 'Duplicate'
+          end
+
+          find('li', text: first_destination_assessment.title).click
+          expect_toastify('Your question has been duplicated.')
+
+          wait_for_page
```

Comment:
`wait_for_page` is unnecessary, don't include it unless it's actually needed since it will cause our tests duration to lengthen.

Reply:
done